moveit_core/robot_model/src/joint_model.cpp

```diff
@@ -118,7 +118,7 @@ bool JointModel::satisfiesVelocityBounds(const double* values, const Bounds& oth
 {
   for (std::size_t i = 0; i < other_bounds.size(); ++i)
   {
-    if (other_bounds[i].max_velocity_ + margin < values[i])
+    if (other_bounds[i].max_velocity_ < values[i] + margin)
```

Comment:
I would prefer this version
```suggestion
    if (other_bounds[i].max_velocity_ - margin < values[i])
```

Reply:
Hi @AndyZe and @henningkayser ,
let me know if I should go ahead with this change?
or 
wait and try to work on removing the `margin` argument completely?

Reply:
It would be great if you could work on removing `margin` arguments completely. I would just do it in this same PR and we can rename the PR.

I expect there will be some CI failures when the change is made. (In fact, I tried it myself and saw the CI failures). But it will be good to see these issues get exposed.

Reply:
@SuyashSapkal here is a list of all usage of `margin` I could find in the codebase. I found this with `grep -r margin`. It's a pretty big list, let me know if you're up for it.

[use_of_margin.txt](https://github.com/ros-planning/moveit2/files/11215979/use_of_margin.txt)

Reply:
ok will work on this and get back